servers/dsub/jobs/controllers/jobs_controller.py

```diff
@@ -179,6 +190,9 @@ def _generate_dstat_jobs(provider, query, create_time_max=None,
     last_create_time = None
     job_buffer = []
     for job in jobs:
+        if query.end and job['end-time'] > query.end:
```

Comment:
Wouldn't this fail for running jobs, i.e. without an end time?

Reply:
Yup, fixed.